db/migrate/20160901013042_change_column_type.rb

```diff
@@ -0,0 +1,14 @@
+class ChangeColumnType < ActiveRecord::Migration
+  def change
+    connection.execute(%q{
+        alter table shares
+        alter column lat
+        type float using lat::float
+    })
+    connection.execute(%q{
+        alter table shares
+        alter column lon
+        type float using lon::float
+    })
```

Comment:
Oh interesting.  Rails used to behave differently (or does behave differently with other databases).  I think it would seamlessly go the other way (float -> string), but it tries to prevent lossy data.  The only other way is probably to remove the column and re-add it, but this is better than that.